src/main/java/net/javadiscord/javabot/events/MessageLinkListener.java

```diff
@@ -0,0 +1,63 @@
+package net.javadiscord.javabot.events;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.GuildChannel;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.javadiscord.javabot.Bot;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MessageLinkListener extends ListenerAdapter {
+
+	private final Pattern MESSAGE_URL_PATTERN = Pattern.compile("https://((?:canary|ptb)\\.)discord.com/channels/[0-9]+/[0-9]+/[0-9]+");
```

Comment:
Discord links can omit the `canary.`/`ptb.` if neither a canary nor PTB is used. It should be `((?:canary|ptb)\.)?` (with a question mark at the end).